scripts/persist_pinocchio.py

```diff
@@ -1,26 +1,63 @@
-from sumtool.storage import store_model_summaries
+from sumtool.storage import store_model_summaries, get_summaries
 import json
 
 
 if __name__ == "__main__":
-    sums_by_id = {}
+    sums_by_id_fallback_ours = {}
+    sums_by_id_fallback_theirs = {}
+    sums_by_id_with_skip = {}
     n_missing = 0
+    n_pinocchio_zero = 0
+    baseline_sums = get_summaries("xsum", "facebook-bart-large-xsum")
     with open("./data/results-pinocchio/xsum-pinocchio-results.jsonl", "r") as f:
         for line in f:
             obj = json.loads(line)
-            summary = obj["pinocchio_predicted"].strip()
-            if summary == "":
+            sum_id = obj["id"]
+            fallback = False
+            pinocchio_summary = obj["pinocchio_predicted"].strip()
+
+            if pinocchio_summary == "":
                 n_missing += 1
-                summary = obj["bart_predicted"].strip()
-            sums_by_id[obj["id"]] = summary
-    print(f"{n_missing} reverted to baseline")
+                fallback = True
+            elif obj["num_pinocchio_fires"] == 0:
+                n_pinocchio_zero += 1
+                fallback = True
+
+            if fallback:
+                sums_by_id_fallback_theirs[sum_id] = obj["bart_predicted"].strip()
+                sums_by_id_fallback_ours[sum_id] = baseline_sums[sum_id]["summary"]
+            else:
+                sums_by_id_fallback_theirs[sum_id] = pinocchio_summary
+                sums_by_id_fallback_ours[sum_id] = pinocchio_summary
+                sums_by_id_with_skip[sum_id] = pinocchio_summary
+
+    print(f"{n_missing} missing reverted to baseline")
+    print(f"{n_pinocchio_zero} with pinocchio fire 0 reverted to baseline")
+    store_model_summaries(
+        "xsum",
+        "pinocchio-fallback-ours",
```

Comment:
@dleve123 For evaluate_summaries we use `pinocchio_fallback_ours` which falls back to our baseline generated summay instead of their baseline generated summery. As evident from the small difference in Rouge-scores (ours is slightly higher) I think this is OK to do. 

It's arguably more fair and saves us from annotating their baseline summaries.